Formula/grokj2k.rb

```diff
@@ -77,23 +73,30 @@ def install
               "\\1 #{Formula["little-cms2"].opt_include}"
     end
 
-    # Workaround to fix build when using Homebrew little-cms2 headers with C++17.
-    # lcms2.h:1279:44: error: ISO C++17 does not allow 'register' storage class specifier
-    # See https://github.com/GrokImageCompression/grok/issues/241
-    ENV.append "CXXFLAGS", "-DCMS_NO_REGISTER_KEYWORD=1"
-
     perl = DevelopmentTools.locate("perl")
+    perl_archlib = Utils.safe_popen_read(perl.to_s, "-MConfig", "-e", "print $Config{archlib}")
+    args = %W[
+      -DGRK_BUILD_DOC=ON
+      -DGRK_BUILD_LIBPNG=OFF
+      -DGRK_BUILD_LIBTIFF=OFF
+      -DPERL_EXECUTABLE=#{perl}
+      -DCMAKE_INSTALL_BINDIR=libexec/bin
+    ]
+    # We install binaries into libexec/bin since we need to create wrapper env scripts
+    # and there is a risk of breakage if we manually move the binaries afterward.
+    # Since pkg-config files are created based on CMAKE_INSTALL_BINDIR, we need to overwrite
+    # these so that users get the path to env script that enables exiftool features.
+    inreplace ["src/lib/core/libgrokj2k.pc.cmake.in", "src/lib/codec/libgrokj2kcodec.pc.cmake.in"],
+              "@CMAKE_INSTALL_BINDIR@", "bin"
+
     if OS.mac?
       # Workaround Perl 5.18 issues with C++11: pad.h:323:17: error: invalid suffix on literal
-      ENV.append "CXXFLAGS", "-Wno-reserved-user-defined-literal" if MacOS.version <= :catalina
+      args << "-DCMAKE_CXX_FLAGS=-Wno-reserved-user-defined-literal" if MacOS.version <= :catalina
```

Comment:
Note that using `CMAKE_CXX_FLAGS` will cause CMake to ignore any `CXXFLAGS` set in the environment.